balancer/rls/balancer.go

```diff
@@ -303,7 +303,10 @@ func (b *rlsBalancer) UpdateClientConnState(ccs balancer.ClientConnState) error
 		// `resizeCache` boolean) because `cacheMu` needs to be grabbed before
 		// `stateMu` if we are to hold both locks at the same time.
 		b.cacheMu.Lock()
-		b.dataCache.resize(newCfg.cacheSizeBytes)
+		evicted := b.dataCache.resize(newCfg.cacheSizeBytes)
+		if evicted {
```

Comment:
Does a change in the cache size effect the picker? If not, why update the picker?

I'm asking since there was no picker update here before this PR.

Reply:
Just read the attached issue: https://github.com/grpc/grpc-go/issues/7212

Cache eviction requires picker re-generation. Ignore this thread also.